src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/AddImports/IAddImportsService.cs

```diff
@@ -41,14 +48,22 @@ private static bool CanAddImportsInHiddenRegions(Document document)
             var spanMapper = document.Services.GetService<ISpanMappingService>();
             return spanMapper != null && spanMapper.SupportsMappingImportDirectives;
         }
+#else
+        public static async Task<AddImportPlacementOptions> FromDocumentAsync(Document document, CancellationToken cancellationToken)
+        {
+            var service = document.GetRequiredLanguageService<IAddImportsService>();
+            var options = document.Project.AnalyzerOptions.GetAnalyzerOptionSet(await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false), cancellationToken);
+            return new(
+                PlaceSystemNamespaceFirst: options.GetOption(GenerationOptions.PlaceSystemNamespaceFirst),
+                PlaceImportsInsideNamespaces: service.PlaceImportsInsideNamespaces(options),
+                AllowInHiddenRegions: false);
```

Comment:
Can we not call `FromDocument` here just like done in the first method in this file?

```csharp
    => FromDocument(document, document.Project.AnalyzerOptions.GetAnalyzerOptionSet(await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false), cancellationToken));
```

Reply:
I am wondering if we can remove all the #if directives in this file and just move it inside the `FromDocumentAsync` method to conditionally compute options, and rest of the code in this file can be available in both layers.